pkg/controllermanager/controller/utils/associations.go

```diff
@@ -92,3 +93,25 @@ func DetermineSecretBindingAssociations(quota *gardenv1beta1.Quota, bindingListe
 	}
 	return associatedBindings, nil
 }
+
+// DetermineBackupBucketAssociations gets a <backupBucketLister> to determine the BackupBucket resources which are associated
+// to given <cloudProfile>
+func DetermineBackupBucketAssociations(cloudProfile *gardenv1beta1.CloudProfile, backupBucketLister gardencorelisters.BackupBucketLister) ([]string, error) {
+	var associatedBackupBuckets []string
+	if cloudProfile.Spec.Backup == nil {
+		return associatedBackupBuckets, nil
+	}
+	backupBuckets, err := backupBucketLister.List(labels.Everything())
+	if err != nil {
+		logger.Logger.Info(err.Error())
+		return nil, err
+	}
+
+	for _, backupBucket := range backupBuckets {
+		// TOCHK
```

Comment:
What shall be checked here? It looks correct, although I would verify that `cloudProfile.Spec.Backup.Region` is not `nil`.

Reply:
Nope. The actual check was against `secretRef` instead of `region`. And i had to look back this cause i was thinking to use alternative to parse cloudProfileName from backupBucketName and determine association.Sorry missed that. Will fix it.